pkg/handler/v1/provider/openstack.go

```diff
@@ -232,6 +239,8 @@ func DecodeOpenstackReq(_ context.Context, r *http.Request) (interface{}, error)
 	req.Tenant = r.Header.Get("Tenant")
 	req.TenantID = r.Header.Get("TenantID")
 	req.Domain = r.Header.Get("Domain")
+	req.ApplicationCredentialID = r.Header.Get("ApplicationCredentialID")
```

Comment:
Request. I think this is fine too, but can you check if that is valid?
I don't think there are any headers required. The differentiation btw username/password and appcredID/Secret can be omitted here.